ddtrace/tracer/span.go

```diff
@@ -135,6 +143,39 @@ func (s *span) setTagError(value interface{}, debugStack bool) {
 	}
 }
 
+// takeStacktrace takes stacktrace
+// skips the stack frame for this function and for the call to runtime.Caller and to the caller -> setTagError
+// also skips the last frame which is runtime
+func takeStacktrace(n, offset uint) string {
+	var builder strings.Builder
+	pcs := make([]uintptr, n)
+
+	i := 0
+	numFrames := runtime.Callers(2+int(offset), pcs)
+	frames := runtime.CallersFrames(pcs[:numFrames])
```

Comment:
I don't remember exactly how this works, but do we have to start from 1 here? I mean `pcs[1:numFrames]` to exclude the call to `takeStacktrace`? I'm just trying to make sure that when `offset` is `0` the first few lines of the stack are exactly the same as `debug.Stack()`

Reply:
That was probably in the previous version of MR and is no longer relevant.

Reply:
> Lastly, can you please submit additional commits for each review round. It would be nice to compare previous versions to newer versions you are making. We can then squash at the very end. Unfortunately with Github it's not possible to go back to previous revisions without having multiple commits in a PR.

I will admit, I don't know how to do that. Can you help me? I am used to ammend all the commits because that's what we do in our team and I forgot not to do it elsewhere.

Reply:
What do you mean by MR? I assume you mean PR. I'm curious what system you are using :)

> Otherwise, with offset 0 the first frame will be setTagError function.

You're right! Because of the +2, right? This is fine then 👍

Reply:
> What do you mean by MR? 

We are using Gitilab internally with Merge Requests 🙂

Reply:
I see :) That explains it!